rfcs/0000-snarkworker-socket.md

```diff
@@ -0,0 +1,64 @@
+## Summary
+[summary]: #summary
+
+Pools of Snarkworkers can be made more robust and be load balanced by moving to a persistant socket connection for work requests and responses.
+
+## Terms
+
+snark-coordinator (SC) - daemon(s) participating on the gossip network and queueing up snark-work to be done by snark-workers
+
+snark-worker (SW) - daemon(s) processing compute intensive snarks (snark+libsnark)
+
+
+## Motivation
+
+[motivation]: #motivation
+
+
+Current snark-worker (SW) to snark-coordinator (SC) communications are done with two discrete communication steps.
+1. SW connect to SC on rpc port and request some work.
+(SW does work.)
+2. SW again connect to SC on rpc port and sends completed work.
+
+If a SW requests work and then dies, the work is NOT re-assigned as the SC is just stuck waiting for the completed work.
```

Comment:
@deepthiskumar says this isn't the case [here](https://github.com/CodaProtocol/coda/issues/1801#issuecomment-474515952).